.github/workflows/_prepare-all.yml

```diff
@@ -0,0 +1,20 @@
+name: Prepare All
+
+on:
+  workflow_call:
+
+jobs:
+  prepare-all:
+    runs-on: macos-latest # use [ self-hosted, macOS ] to host on our own mac mini, which is twice as fast and cheaper than hosting on a github runner. See README for more info.
```

Comment:
Price of Mac runners is 10x that of linux: https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#minute-multipliers

What is your reason for choosing mac? Did you try running on linux? If not, I would say it is at least worth the effort to do a test run.

edit: it tried and it failed. For an unexpected reason: it couldn't find the requested version of firebase
Up to you whether you feel this is worth looking into.

Reply:
Could be related to java version incompatibility somehow, even though java 21 is mentioned: https://github.com/firebase/firebase-android-sdk/issues/6123

Reply:
Because we use our own mac-mini on our projects, that is free. Switching from mac -> mac-mini is no effort.

I added documentation on how to use our own mac-mini. In the "Using this template" step, "You probably want to enable the self-hosted runner". I'll update it a bit to be even more clear.